odrive-motor/src/odriveCAN.py

```diff
@@ -0,0 +1,259 @@
+        try:
+            bus.send(msg)
+        except can.CanError:
+            pass
+
+        def periodically_surface_errors(odriveCAN):
+            while True:
+                asyncio.run(odriveCAN.surface_errors())
+                time.sleep(1)
+
+        thread = Thread(target = periodically_surface_errors, args=[odriveCAN])
+        thread.setDaemon(True) 
+        thread.start()
+
+        return odriveCAN
+    
+    @classmethod
+    def validate(cls, config: ComponentConfig):
+        return
+
+    async def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):
+        baud_rate = "250000"
+        if config.attributes.fields["canbus_baud_rate"].string_value != "":
+            baud_rate = config.attributes.fields["canbus_baud_rate"].string_value
+            baud_rate = baud_rate[0:len(baud_rate)-1] + "000"
+
+        if baud_rate != self.baud_rate:
+            self.baud_rate = baud_rate
+            Popen("sudo ip link set can0 down", shell=True).wait()
+            Popen("sudo ip link set can0 up type can bitrate " + str(baud_rate), shell=True).wait()
+            # send an arbitrary message because the first message sent over CAN does not go through
+            msg = db.get_message_by_name('Clear_Errors')
+            msg = can.Message(arbitration_id=msg.frame_id | self.nodeID << 5, is_extended_id=False)
+            try:
+                bus.send(msg)
+            except can.CanError:
+                pass
+        
+        new_nodeID = config.attributes.fields["canbus_node_id"].number_value
+        if new_nodeID != self.nodeID:
+            self.set_node_id(new_nodeID)
+
+    async def set_power(self, power: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        torque = power*self.current_limit*self.torque_constant
+        self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+        await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
+        self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x01, 'Input_Mode': 0x01})
+        self.send_can_message('Set_Input_Torque', {'Input_Torque': torque})
+
+    async def go_for(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        rps = rpm / MINUTE_TO_SECOND
+
+        self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+        await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
+
+        if revolutions == 0.0:
+            self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x02, 'Input_Mode': 0x01})
+            self.send_can_message('Set_Input_Vel', {'Input_Vel': rps, 'Input_Torque_FF': 0})
+
+        else:
+            self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
+            self.send_can_message('Set_Limits', {'Velocity_Limit': 10.0, 'Current_Limit': 40.0})
+
+            current_position = await self.get_position()
```

Comment:
I have no idea why but using `go_for` with revolutions > 0 causes major issues and I have to restart the server but everything else works perfectly find so I need to investigate that more

Reply:
even GoTo?